dotnet/src/Connectors/Connectors.AzureAIInference/Extensions/AzureAIInferenceKernelBuilderExtensions.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using Azure.AI.Inference;
+using Microsoft.SemanticKernel.Connectors.AzureAIInference;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Provides extension methods for <see cref="IKernelBuilder"/> to configure Azure AI Inference connectors.
+/// </summary>
+public static class AzureAIInferenceKernelBuilderExtensions
+{
+    /// <summary>
+    /// Adds the <see cref="AzureAIInferenceChatCompletionService"/> to the <see cref="IKernelBuilder.Services"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IKernelBuilder"/> instance to augment.</param>
+    /// <param name="modelId">Target Model Id for endpoints supporting more than one model</param>
+    /// <param name="apiKey">API Key</param>
+    /// <param name="endpoint">Endpoint / Target URI</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    /// <param name="serviceId">A local identifier for the given AI service</param>
+    /// <returns>The same instance as <paramref name="builder"/>.</returns>
+    public static IKernelBuilder AddAzureAIInferenceChatCompletion(
+        this IKernelBuilder builder,
+        string? modelId = null,
+        string? apiKey = null,
```

Comment:
We also need to support using a `TokenCredential` as we will have to disable use of API keys as part of security wave 2.